compiler/res/prog8lib/atari/textio.p8

```diff
@@ -118,7 +118,28 @@ asmsub  scroll_down  (ubyte alsocolors @ Pc) clobbers(A)  {
 }
 
 
-romsub $FFD2 = chrout(ubyte char @ A)    ; TODO
+romsub $F2B0 = outchar(ubyte char @ A)
+romsub $F2Fd = waitkey()
+
+asmsub chrout(ubyte char @ A) {
+	%asm {{
+		sta  _tmp_outchar+1
+		pha
+		txa
+		pha
+		tya
+		pha
+_tmp_outchar
+		lda  #0
+		jsr  outchar
+		pla
+		tay
+		pla
+		tax
+		pla
+		rts
+	}}
+}
```

Comment:
as print() uses chrout(), printing a string will be very slow from what I can see now. Does the atari kernal perhaps provide a true print() routine itself?

Reply:
Yes, there are a lot more options and OS calls, but I just wanted to get something working ;)
Would be better to use the E: handler and CIO calls to print characters to the screen.

Reply:
That's fair, we can take this step by step.